### PR TITLE
feat(web-analytics): Add page-level conversion goals, part 1

### PR DESCRIPTION
**Pull request by robbie-c**
## Problem

Conversion goals are an often-requested feature of web analytics.

## Changes

Added a button to add conversion goals
Screenshot 2024-09-11 at 12 14 04

Uses Actions: 
Screenshot 2024-09-11 at 12 15 11

If a conversion goal is set, the overview shows conversion-related metrics
Screenshot 2024-09-11 at 12 19 08

If a conversion goal is set, the graphs tile shows conversion-related metrics
Screenshot 2024-09-11 at 12 17 52

Screenshot 2024-09-11 at 12 17 25

Still to do (not in this PR, in a follow-up PR):
* Make the stats table queries (e.g. the Path queries) calculate conversion rate instead of bounce rate if a goal is set
* Make the conversion goal chooser UI match the filter chooser UI, rather than the just being the default action picker UI
* Make the goals clickable in the goals query

## Does this work well for both Cloud and self-hosted?

Yes

## How did you test this code?
Added some tests for the new query

### CODE REVIEW

**Review by andyvan-ph — APPROVED** [review]

File: frontend/src/scenes/web-analytics/webAnalyticsLogic.tsx [hunk 1]
```diff
@@ -467,6 +472,12 @@ export const webAnalyticsLogic = kea<webAnalyticsLogicType>([
                 setShouldStripQueryParams: (_, { shouldStripQueryParams }) => shouldStripQueryParams,
             },
         ],
+        conversionGoal: [
+            null as WebAnalyticsConversionGoal | null,
```
Comment (daibhin) [inline frontend/src/scenes/web-analytics/webAnalyticsLogic.tsx:476]: Going with the above suggestion would make this null redundant
```suggestion
            null as WebAnalyticsConversionGoal,
```

**Review by daibhin — APPROVED** [review]
Couple of small things but looks good 👍

File: frontend/src/queries/schema.ts [hunk 2]
```diff
@@ -1330,10 +1330,13 @@ export interface SessionsTimelineQuery extends DataNode<SessionsTimelineQueryRes
...
 export type WebAnalyticsPropertyFilter = EventPropertyFilter | PersonPropertyFilter | SessionPropertyFilter
 export type WebAnalyticsPropertyFilters = WebAnalyticsPropertyFilter[]
-
+export type WebAnalyticsConversionGoal = {
+    actionId: integer
+}
 interface WebAnalyticsQueryBase<R extends Record<string, any>> extends DataNode<R> {
     dateRange?: DateRange
     properties: WebAnalyticsPropertyFilters
+    conversionGoal?: WebAnalyticsConversionGoal
```
[the remark this answers is not in the document]
Comment (robbie-c) [reply frontend/src/queries/schema.ts]: Done

File: frontend/src/scenes/web-analytics/webAnalyticsLogic.tsx [hunk 3]
```diff
@@ -1594,6 +1748,40 @@ export const webAnalyticsLogic = kea<webAnalyticsLogicType>([
...
+                }
+            } else {
+                if (
+                    tab === GraphsTab.TOTAL_CONVERSIONS ||
+                    tab === GraphsTab.CONVERSION_RATE ||
+                    tab === GraphsTab.UNIQUE_CONVERSIONS
+                ) {
+                    actions.setGraphsTab(GraphsTab.UNIQUE_USERS)
+                }
+            }
```
[the remark this answers is not in the document]
Comment (robbie-c) [reply frontend/src/scenes/web-analytics/webAnalyticsLogic.tsx]: Done

File: frontend/src/scenes/web-analytics/webAnalyticsLogic.tsx [hunk 4]
```diff
@@ -557,114 +576,240 @@ export const webAnalyticsLogic = kea<webAnalyticsLogicType>([
...
+                                                  filterTestAccounts,
+                                                  properties: webAnalyticsFilters,
+                                              },
+                                              hidePersonsModal: true,
+                                              embedded: true,
+                                          },
+                                          showIntervalSelect: true,
+                                          insightProps: createInsightProps(TileId.GRAPHS, GraphsTab.UNIQUE_USERS),
+                                          canOpenInsight: true,
+                                      }
```
[the remark this answers is not in the document]
Comment (robbie-c) [reply frontend/src/scenes/web-analytics/webAnalyticsLogic.tsx:809]: I'll likely do this as a follow-up

File: frontend/src/scenes/web-analytics/WebConversionGoal.tsx [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+export const WebConversionGoal = (): JSX.Element | null => {
+    const { conversionGoal } = useValues(webAnalyticsLogic)
+    const { featureFlags } = useValues(featureFlagLogic)
+    const { setConversionGoal } = useActions(webAnalyticsLogic)
+    const { actions } = useValues(actionsModel)
+
+    if (!featureFlags[FEATURE_FLAGS.WEB_ANALYTICS_CONVERSION_GOALS] && !conversionGoal) {
+        return null
+    }
```
[the remark this answers is not in the document]
Comment (robbie-c) [reply frontend/src/scenes/web-analytics/WebConversionGoal.tsx]: Done

File: frontend/src/scenes/web-analytics/WebPropertyFilters.tsx [hunk 6]
```diff
@@ -12,16 +12,18 @@ export const WebPropertyFilters = ({
...
-                TaxonomicFilterGroupType.EventProperties,
-                TaxonomicFilterGroupType.PersonProperties,
-                TaxonomicFilterGroupType.SessionProperties,
-            ]}
-            onChange={(filters) => setWebAnalyticsFilters(filters.filter(isEventPersonOrSessionPropertyFilter))}
-            propertyFilters={webAnalyticsFilters}
-            pageKey="web-analytics"
-            eventNames={['$pageview']}
-        />
+        <>
```
[the remark this answers is not in the document]
Comment (robbie-c) [reply frontend/src/scenes/web-analytics/WebPropertyFilters.tsx]: Removed

File: frontend/src/scenes/web-analytics/webAnalyticsLogic.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (robbie-c) [reply frontend/src/scenes/web-analytics/webAnalyticsLogic.tsx:809]: I did this here https://github.com/PostHog/posthog/pull/24934

**Pull request merged**